src/gateway_mfr_worker.erl

```diff
@@ -276,7 +276,11 @@ check_miner_key(Pid) ->
         {ok, PubKey} ->
             case ecc_compact:is_compact(PubKey) of
                 {true, _} -> ok;
-                false -> {error, not_compact}
+                false ->                     
+                    case gen_compact_key(Pid, ?KEY_SLOT) of
```

Comment:
This does not seem correct..  check_miner_key is _not_ there to actually generate the private key. It's just there to validate that there is a public compact key in a slot. There is a separate command to generate the private key

Reply:
If check the key not compact, it will always fail when run test. We can't run provision again to re-gen the private key any more. So this code can fix it by re-gen private key.

Reply:
This is not the right assumption. Check_key_configuration should _only_ check that the chip was _already_ correctly configured. If it returns an error on is_compact it means it was _not_ provisioned/configured correctly in a previous step